src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -762,18 +771,19 @@ protected function getServiceBoundManagers(Nette\DI\ServiceDefinition $def)
 
 	public function afterCompile(Code\ClassType $class)
 	{
-		$init = $class->methods['initialize'];
-
-		$init->addBody('Kdyby\Doctrine\Diagnostics\Panel::registerBluescreen($this);');
-
-		if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
-			$blueScreen = 'Tracy\Debugger::getBlueScreen()';
-			$commonDirname = dirname(Nette\Reflection\ClassType::from('Doctrine\Common\Version')->getFileName());
-
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(Nette\Reflection\ClassType::from('Kdyby\Doctrine\Exception')->getFileName())));
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(dirname(dirname(dirname($commonDirname)))))); // this should be vendor/doctrine
-			foreach ($this->proxyAutoloaders as $dir) {
-				$init->addBody($blueScreen . '->collapsePaths[] = ?;', array($dir));
+		if (interface_exists('Tracy\IBarPanel')) {
```

Comment:
This is ugly. I'd suggest moving it to private method and call it conditionally (or always and return conditionally).